ts/main.ts

```diff
@@ -0,0 +1,64 @@
+interface Anime {
+  mal_id: number;
+  title: string;
+  episodes: number;
+  images: {
+    jpg: {
+      image_url: string;
+    };
+  };
+  [key: string]: any; // To account for any additional fields
+}
+
+const animeContainer = document.querySelector(
+  '.anime-container',
+) as HTMLDivElement;
+
+async function fetchAnime(): Promise<Anime[]> {
+  try {
+    const response = await fetch('https://api.jikan.moe/v4/top/anime');
+    const data = await response.json();
+    return data.data as Anime[];
+  } catch (error) {
+    console.error('Error fetching anime data:', error);
+    return [];
+  }
+}
+
+function renderAnime(animeList: Anime[]): void {
+  animeContainer.innerHTML = '';
+
```

Comment:
You generally want to avoid using `innerHTML` because theoretically you could add in data here that represents HMTL elements and it would be a security issue. Do you need to be resetting the `animeContainer` here?